From b85d717c6e03e99ea62717279cd53a4877feb170 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 5 Jul 2017 11:14:54 +0900
Subject: drm: rcar-du: Fix default resolution for Analog RGB

Due to limitation of H/W, the upper limit of dotclock is 50 MHz,
so it is modified from XGA to SVGA in Analog RGB connector.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_vgacon.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
index e81dfbe..8b95569 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
@@ -23,7 +23,15 @@
 
 static int rcar_du_vga_connector_get_modes(struct drm_connector *connector)
 {
-	return drm_add_modes_noedid(connector, 1024, 768);
+	struct drm_device *dev = connector->dev;
+	struct rcar_du_device *rcdu;
+
+	rcdu = dev->dev_private;
+
+	if (rcar_du_has(rcdu, RCAR_DU_FEATURE_D3))
+		return drm_add_modes_noedid(connector, 800, 600);
+	else
+		return drm_add_modes_noedid(connector, 1024, 768);
 }
 
 static const struct drm_connector_helper_funcs connector_helper_funcs = {
-- 
1.9.1

